service/inject_definition.go

```diff
@@ -7,8 +7,8 @@ import (
 	"github.com/mesg-foundation/core/x/xstrings"
 )
 
-// injectDefinition applies service definition to Service type.
-func (s *Service) injectDefinition(def *importer.ServiceDefinition) {
+// InjectDefinition applies service definition to Service type.
+func (s *Service) InjectDefinition(def *importer.ServiceDefinition) {
```

Comment:
This function should not be exported. It's an internal function that changes a lot the data. It should be controlled from inside not outside.
One solution could be to move that directly in the deploy api, the deploy is responsible to populate this service

Reply:
done. I was thinking about doing these improvements later after seeing what service pkg will become to when refactoring is completed. but it's fine